### DISCUSSION TITLE
Cannot write inside a notarized application package

### DISCUSSION THREAD

**Discussion by 42retfa** [airsdk/Adobe-Runtime-Support#2736] [Q&A] [answered]
Hi,

We distribute a macOS application that is able to update itself by downloading and writing .swf files in the resource folder of its packages. this mechanism worked correctly (for many years) until the application was signed and notarized.

` sudo myApp.app/Contents/MacOS/myApp`

It seems that the application is no longer able to download and/or write the update .swf files unless it is run via the sudo 'update' command.  

Any idea to solve this problem.

Thanks in advance.

Vincent.

**Comment by ajwfrost** [comment, marked as answer]
This may be two questions .. the first, if you've got an application that's in the "/Applications" folder, then I would have thought it's likely that only sudo (administrator) would be able to write to folders within the application bundle.

But in terms of code signing and notarisation, you shouldn't add files into the application bundle itself, you should download the SWFs and pick them up from an application storage folder..
https://stackoverflow.com/questions/59350055/how-to-update-a-macos-bundle-without-breaking-notarization

I just looked at a third party bit of software, and adding a file into its 'Resources' folder didn't actually require sudo rights (though nor did it when I tested a .air file that's been installed). But, it did immediately invalidate the code signature. So, best to use app store for these.

I believe this changed when they introduced the idea of "sealed resources" in the code signature a little while back (v20400).

**Comment by 42retfa** [comment]
Hi Andrew,

As always, thank you for the quick answer !
That's what I was afraid of...

Problem with loading the SWFs and pick them up from an application storage folder is that if the main.swf of the app needs an update, the whole application as to be updated. 

I imagine a solution would be to produce an app in which main.swf would be an empty shell responsible for loading the content of the application.

Vincent.

**Reply by ajwfrost** [reply]
Yes, it would mean having the initial swf to just have logic to pick out (and validate?) the appropriate SWF to then run for the main application.. not ideal I know. The alternative though is a full package/bundle update. I think this is just something we have to live with due to the Apple code signing security.

Although - if anyone does have alternative options, we'd be more than happy to try to get them to work with AIR!

thanks
